test/test_index_string.cpp

```diff
@@ -250,6 +251,35 @@ using non_nullable = std::false_type;
 
 } // anonymous namespace
 
+TEST(Tokenizer_Basic)
+{
+    auto tok = realm::Tokenizer::get_instance();
+
+    tok->reset("to be or not to be");
+    auto tokens = tok->get_all_tokens();
+    CHECK_EQUAL(tokens.size(), 4);
+
+    tok->reset("To be or not to be");
+    realm::TokenInfoMap info = tok->get_token_info();
+    CHECK_EQUAL(info.size(), 4);
+    realm::TokenInfo& i(info["to"]);
+    CHECK_EQUAL(i.positions.size(), 2);
+    CHECK_EQUAL(i.positions[0], 0);
+    CHECK_EQUAL(i.positions[1], 4);
+    CHECK_EQUAL(i.ranges.size(), 2);
+    CHECK_EQUAL(i.ranges[0].first, 0);
+    CHECK_EQUAL(i.ranges[0].second, 2);
+    CHECK_EQUAL(i.ranges[1].first, 13);
+    CHECK_EQUAL(i.ranges[1].second, 15);
+
+    tok->reset("Jeg gik mig over sø og land");
+    info = tok->get_token_info();
+    CHECK_EQUAL(info.size(), 7);
+    realm::TokenInfo& j(info["sø"]);
+    CHECK_EQUAL(j.ranges[0].first, 17);
+    CHECK_EQUAL(j.ranges[0].second, 20);
+}
```

Comment:
```suggestion

    tok->reset("with-hyphen -term -other-term-plus");
    CHECK(tok->get_all_tokens() == std::set<std::string>({"with", "hyphen", "term", "other", "plus"}));
}
```

That's essentially how right now tokenizer works. It may not be suitable for usage of parsing search terms in fts with current desing. Quite a few unexpected results.

Reply:
But this is how the index is build today. So you can't find objects with "with-hyphen" included.